### DISCUSSION TITLE
DockerOperator overwriting Docker file ENV vars

### DISCUSSION THREAD

**Discussion by UFMurphy** [apache/airflow#24801] [Q&A] [answered]
Hola,

So here is my situation. I have a basic image I am building where I specify ENV vars in the Dockerfile.

I also want to add additional environment variables from the DockerOperator.

However, the environment variables in the DockerOperator seem to clobber the ones specified in the Dockerfile.

Does anyone know ways to append to the env vars in the Dockerfile via the DockerOperator?

Jim

**Comment by alexkruc** [comment, marked as answer]
Looking at Docker's documentation of the CLI, this is the [required behavior](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) of the `docker run` command :(
Basically, the documentation claims that passing the env vars using runtime, are overwriting the ones set inside the Dockerfile. 
From what I see, the Python API of Docker is mimicking the cli and creates the same behavior.

Unless there is something I'm missing, I don't think that it's an Airflow problem or feature, but Docker's...

**Comment by UFMurphy** [comment]
Thanks, Alex. Much appreciated .
